### PR TITLE
DEVTOOLS-80 Added an interactive mode for create command

### PR DESCRIPTION
**Pull request by fschargorosdky**
* Added errors for create command
 * Existing API name
 * Existing API version name
 * root RAML not found
 * orgId not found
 * apiId not found

### CODE REVIEW

File: lib/factories/createStrategyFactory.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+          'name', 'id', sortByName(businessGroups));
+      },
+      getAPI: function (apis) {
+        return commandPrompt.getChoice(messages.apiPromptMessage(),
+          'name', 'id', sortByName(apis));
+      },
+      getAPIName: function (apis) {
+        return commandPrompt.getInput(messages.apiNamePromptMessage())
+          .then(function (apiName) {
+            return checkAPINameIsNotUsed(apis, apiName)
```
Comment (juancoen) [inline lib/factories/createStrategyFactory.js]: can this be placed inside a tap instead of a then?

File: lib/factories/createStrategyFactory.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+              .return(apiName);
+          });
+      },
+      getAPIVersionName: function (apis, apiId) {
+        return commandPrompt.getInput(messages.apiVersionNamePromptMessage())
+          .then(function (versionName) {
+            return checkAPIVersionNameIsNotUsed(apis, apiId, versionName)
```
Comment (juancoen) [inline lib/factories/createStrategyFactory.js]: same here

File: lib/factories/createStrategyFactory.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                text: 'API',
+                value: true
+              },
+              {
+                text: 'API Version',
+                value: false
+              }
+            ])
+          .then(function (userChoice) {
+            return userChoice.value;
```
Comment (juancoen) [inline lib/factories/createStrategyFactory.js]: can this be replaced with a .get('value'); ?

File: lib/factories/createStrategyFactory.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+          BPromise.reject(new errors.ChoiceNotFoundError(
+            messages.rootRamlDescription()));
+      }
+    };
+  }
+
+  function checkAPINameIsNotUsed(apis, apiName) {
+    var existingAPI = _.find(apis, 'name', apiName.toString());
+    if (existingAPI) {
+      return BPromise.reject(new errors.RepeatedAPINameError(
```
Comment (juancoen) [inline lib/factories/createStrategyFactory.js]: it seems a little strong to make the command fail if the name is repeated. I think a better solution would be to catch the error and prompt the user for a different name

File: lib/factories/createStrategyFactory.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+    return BPromise.resolve();
+  }
+
+  function checkAPIVersionNameIsNotUsed(apis, apiId, versionName) {
+    if (addingAPIVersion(apiId)) {
+      var api = _.find(apis, 'id', apiId);
+      var existingVersion = _.find(api.versions, 'name', versionName.toString());
+      if (existingVersion) {
+        return BPromise.reject(new errors.RepeatedAPIVersionNameError(
```
Comment (juancoen) [inline lib/factories/createStrategyFactory.js]: same here

File: lib/utils/commandPrompt.js [hunk 6]
```diff
@@ -32,8 +34,25 @@ module.exports = function (inquirer) {
     });
   }
 
+  function getRawChoice(message, choices) {
+    if (choices.length > 9) {
+      choices.push(new inquirer.Separator());
+    }
+
+    return new BPromise(function (resolve) {
+      inquirer.prompt({
```
Comment (juancoen) [inline lib/utils/commandPrompt.js]: can this repeated code be avoided?

File: lib/utils/messages.js [hunk 7]
```diff
@@ -400,6 +418,16 @@ module.exports = function () {
     },
     decompressError: function (filePath, error) {
       return 'There was an error decompressing ' + filePath + '\n' + error;
+    },
+    repeatedAPINameError: function (apiId, apiName) {
+      var output = 'Can\'t create API. \n';
+      output += 'There is an API named ' + apiName + ', id: ' + apiId;
+      return output;
+    },
+    RepeatedAPIVersionNameError: function (versionId, versionName) {
```
Comment (juancoen) [inline lib/utils/messages.js]: this method should not start with a capital letter

File: lib/factories/createStrategyFactory.js (hunk 1 above)
Comment (fschargorosdky) [inline lib/factories/createStrategyFactory.js]: done

File: lib/factories/createStrategyFactory.js (hunk 2 above)
Comment (fschargorosdky) [inline lib/factories/createStrategyFactory.js]: done

File: lib/factories/createStrategyFactory.js (hunk 3 above)
Comment (fschargorosdky) [inline lib/factories/createStrategyFactory.js]: done

File: lib/factories/createStrategyFactory.js (hunk 4 above)
Comment (fschargorosdky) [inline lib/factories/createStrategyFactory.js]: done

File: lib/factories/createStrategyFactory.js (hunk 5 above)
Comment (fschargorosdky) [inline lib/factories/createStrategyFactory.js]: done

File: lib/utils/commandPrompt.js (hunk 6 above)
Comment (fschargorosdky) [inline lib/utils/commandPrompt.js]: done

File: lib/utils/messages.js (hunk 7 above)
Comment (fschargorosdky) [inline lib/utils/messages.js]: done

**Comment by juancoen** [comment]
looks good to me

**Pull request merged**